MANIFEST.in

```diff
@@ -1,3 +1,4 @@
-include LICENSE
-include README.md
-include requirements.txt
+global-exclude *.png *.gif
+prune dashboard
+prune menubar
+prune .github
```

Comment:
I wouldn't care about the text files in this folder. There's no blobs there, right?

Reply:
Yes, no blobs.  But I removed it with following thought process:

1) I haven't removed anything which will prohibit `sdist` users from building/running/testing/docgen locally.  All of that is still possible.
2) So `.github` workflows become redundant.  And if a user wants to use GHA, they must instead fork the repo.  As most likely, these workflows can/may change.

Reply:
I know but maintaining a long list of entries here results in a maintenance burden which is something that is good to avoid whenever possible. sdist has "source" in its name and the content should be as close to its definition. Besides, being a source of truth offline, the workflows have invocation examples for different things that are not explicitly documented.
Another use is including some files in the Sphinx docs.

Reply:
> the workflows have invocation examples for different things that are not explicitly documented

Correct.  Lemme get the `.github` flows in for now.  Frankly, I want to keep `Makefile` as source of truth for all invocations.  So in future, we might move those invocations in `Makefile` and let `tox` use make.

> I know but maintaining a long list of entries here results in a maintenance burden which is something that is good to avoid whenever possible.

Naa, I don't expects files to be added/removed here.  Current GitHub repo is a collection of things right.  It contains the library (`proxy.py`, `examples`, `tests`) and other stuff (`dashboard`, `menubar`).   `helper` is special case which contains utilities also useful at runtime (monitoring) and in benchmarks.  I'll eventually move the benchmark stuff in it's own dedicated top-level directory.

So question remains, what really is the source here?   I think the library bits only.  As we discussed earlier, we'll anyways move `dashboard` out of this repo.  Similarly, `menubar`, `benchmark` will have their own separate repos.  Hence, I feel contents related to library only should be shipped.

Reply:
> > the workflows have invocation examples for different things that are not explicitly documented
> 
> Correct. Lemme get the `.github` flows in for now. Frankly, I want to keep `Makefile` as source of truth for all invocations. So in future, we might move those invocations in `Makefile` and let `tox` use make.

This is a very bad idea because make doesn't have any context into the Python world, nor is it aware of how to cache virtualenvs properly. It does not ensure the isolations between different envs at all being exposed to random and uncontrolled external influence. You could look into `nox`, if you want something configured via scripting but make is totally unfit for the task.

Also, many downstream packaging ecosystems have macros for working with tox-driven projects. They will (semi-)automatically pick up testing and building the docs. And they'll work with the PEP 517 setup seamlessly. Don't take this away from them.

> > I know but maintaining a long list of entries here results in a maintenance burden which is something that is good to avoid whenever possible.
> 
> Naa, I don't expects files to be added/removed here. Current GitHub repo is a collection of things right. It contains the library (`proxy.py`, `examples`, `tests`) and other stuff (`dashboard`, `menubar`). `helper` is special case which contains utilities also useful at runtime (monitoring) and in benchmarks. I'll eventually move the benchmark stuff in it's own dedicated top-level directory.
> 
> So question remains, what really is the source here? I think the library bits only. As we discussed earlier, we'll anyways move `dashboard` out of this repo. Similarly, `menubar`, `benchmark` will have their own separate repos. Hence, I feel contents related to library only should be shipped.

When you'll move those away, there will be almost nothing to exclude (for example, https://github.com/pytest-dev/pytest, https://github.com/cherrypy/cheroot, and many other high-profile ones). Most of the projects don't even bother to exclude the web-only images (well, this is also because they've switched to a more optimal SVG format that is Git-friendly because it's text unlike bitmap images that are blobs).

I think you're also at risk of excluding too much which may eventually hurt the downstream packagers.